docs/doc_4_authentication.md

```diff
@@ -40,3 +39,160 @@ func main() {
 	fmt.Printf("Projects: %v\n", projects)
 }
 ```
+
+## (Preview) Using the Atlas Go SDK with Service Account Authentication
+
+Atlas SDK Go provides OAuth Authentication using Service Accounts (currently available as a [Preview](https://www.mongodb.com/resources/beta-programs) feature)
+Service Account implement OAuth [client_credentials](https://oauth.net/2/grant-types/client-credentials) grant.
+For more information about feature please refer to [Service Account Public documentation.](https://www.mongodb.com/docs/atlas/api/service-accounts-overview/)
+
+## OAuth Authentication
+
+### Authenticating with OAuth ClientID and ClientSecret
+
+### Admin API Authentication using Service Accounts
+
+Authenticating using Service Accounts
+
+```go
+package main
+
+import (
+	"log"
+	"os"
+
+	"go.mongodb.org/atlas-sdk/v20241023001/admin"
+)
+
+func main() {
+	clientID := os.Getenv("MONGODB_ATLAS_CLIENT_ID")
+	clientSecret := os.Getenv("MONGODB_ATLAS_CLIENT_SECRET")
+
+	if clientID == "" || clientSecret == "" {
+		log.Fatal("Missing CLIENT_ID or CLIENT_SECRET environment variables")
+	}
+
+	// Using ClientID and ClientSecret. No cache supported (nil).
+	sdk, err := admin.NewClient(admin.UseOAuthAuth(clientID, clientSecret, nil))
+
+	// Make API calls
+}
+```
+
+> NOTE: This method of initialization provides **No Ability to Revoke API**: once a token is issued, it cannot be invalidated until it expires.
+
+### Specifying a Token Cache
+
+In this example, we will demonstrate how to use the OAuth Client Credentials flow with a custom token cache.
+The cache allows you to store OAuth tokens for reuse across application restarts, improving efficiency by reducing the number of token requests to the authorization server.
+
+**Note:** Cache prevents from requesting multiple tokens which are subject to OAuth Token Limits and Rate Limiting.
```

Comment:
```suggestion
The cache allows you to store OAuth tokens for reuse across application restarts, improving efficiency by reducing the number of token requests to the authorization server. This reduction in requests also minimizes the impact of OAuth Token Limits and Rate Limiting.
```

I think the word 'minimizing' is important here; we probably can't provide a 100% guarantee that the user won't ever run afoul of these limits, so we should make it clear that while this sharply mitigates the concern, it doesn't eliminate it. Also, folding that statement into the existing paragraph spares us some redundancy in the note.